dashboard/serializers.py

```diff
@@ -15,13 +20,15 @@ class UserProgramSearchSerializer:
     Provides functions for serializing a ProgramEnrollment for the ES index
     """
     @staticmethod
-    def calculate_certificate_grade_average(certificates):
+    def calculate_final_grade_average(mmtrack):
         """
-        Calculates an average grade (integer) from a list of certificates with <1 decimal grades
+        Calculates an average grade (integer) from the program final grades
         """
-        return None if len(certificates) == 0 else round(
-            (sum((Decimal(certificate['grade']) for certificate in certificates)) / len(certificates)) * 100
-        )
+        if not mmtrack.course_ids:
+            return
```

Comment:
looks like this check is unnecessary. if `course_ids` is blank in this object, `get_all_final_grades` will just return an empty dict. you should just add an `else` clause below and return None there